test/NMSConnectionFactoryTest.cs

```diff
@@ -89,6 +68,56 @@ public void TestURI(string connectionURI)
 			}
 		}
 
+        [TestCase("activemq:tcp://${activemqhost}:61616/InvalidHost:0")]
+		[TestCase("activemq:tcp://${activemqhost}:61616/0.0.0.0:-1")]
+		[TestCase("tcp://InvalidHost:61616")]
+		[TestCase("activemq:tcp://InvalidHost:61616")]
+		[TestCase("activemq:tcp://InvalidHost:61616?connection.asyncclose=false")]
+
+		[TestCase("tcp://${activemqhost}:61616?connection.InvalidParameter=true")]
+		[TestCase("activemq:tcp://${activemqhost}:61616?connection.InvalidParameter=true")]
+		[TestCase("activemq:failover:tcp://${activemqhost}:61616?connection.InvalidParameter=true")]
+		[TestCase("activemq:failover:(tcp://${activemqhost}:61616)?connection.InvalidParameter=true")]
+		[TestCase("activemq:failover:(tcp://${activemqhost}:61616,tcp://${activemqbackuphost}:61616)?connection.InvalidParameter=true")]
+
+		[TestCase("ftp://${activemqhost}:61616")]
+		[TestCase("http://${activemqhost}:61616")]
+		[TestCase("discovery://${activemqhost}:6155")]
+		[TestCase("sms://${activemqhost}:61616")]
+		[TestCase("activemq:multicast://${activemqhost}:6155")]
+		[TestCase("activemq:(tcp://${activemqhost}:61616)?connection.asyncClose=false")]
+        public void TestURIFailures_NMSConnectionException(string connectionURI)
+        {
+            Assert.Throws<NMSConnectionException>(() =>
+            {
+                NMSConnectionFactory factory = new NMSConnectionFactory(NMSTestSupport.ReplaceEnvVar(connectionURI));
+                Assert.IsNotNull(factory);
+                Assert.IsNotNull(factory.ConnectionFactory);
+                using(IConnection connection = factory.CreateConnection("", ""))
+                {
+                    Assert.IsNotNull(connection);
+                    connection.Close();
+                }
+            });
+        }
+
+        [TestCase("(tcp://${activemqhost}:61616,tcp://${activemqhost}:61616)")]
+        [TestCase("tcp://${activemqhost}:61616,tcp://${activemqhost}:61616")]
+        public void TestURIFailures_UriFormatException(string connectionURI)
```

Comment:
Because `ExpectedException` was deprecated in NUnit3, I split up these tests from the ones above. They expect a different exception to be thrown and it was easier to break them up than re-create the ExpectedException logic in the `[TestCase]` attribute.

Reply:
Lets look to remove, as we did with API. If you look at the change history youll see how tests there we rewritten with ease